packages/tc-api-db-manager/__tests__/db-manager.spec.js

```diff
@@ -60,37 +78,91 @@ describe('creating db constraints', () => {
 	});
 
 	it("doesn't create a uniqueness constraint if it does exist", async () => {
-		mockConstraints(dbRun, [
-			'CONSTRAINT ON (s:Dog) ASSERT s.nose IS UNIQUE',
-		]);
+		mockConstraints({
+			stub: dbRun,
+			constraints: ['CONSTRAINT ON (s:Dog) ASSERT s.nose IS UNIQUE'],
+		});
 		schema.getTypes.mockReturnValue([
 			{ name: 'Dog', properties: { nose: { unique: true } } },
 		]);
 		await initConstraints();
 		expect(dbRun).not.toHaveBeenCalledWith(
-			'CREATE CONSTRAINT ON (s:Dog) ASSERT s.tail IS UNIQUE',
+			'CREATE CONSTRAINT ON (s:Dog) ASSERT s.nose IS UNIQUE',
 		);
+		expect(dbRun).toHaveBeenCalledTimes(4);
 	});
 
 	// ignore this test until we upgrade to enterprise adiition - community edition doesn't have exists()
 	it.skip("doesn't create an existence constraint if it does exist", async () => {
-		mockConstraints(dbRun, ['CONSTRAINT ON (s:Dog) ASSERT exists(s.nose)']);
+		mockConstraints({
+			stub: dbRun,
+			constraints: ['CONSTRAINT ON (s:Dog) ASSERT exists(s.nose)'],
+		});
 		schema.getTypes.mockReturnValue([
 			{ name: 'Dog', properties: { nose: { required: true } } },
 		]);
 		await initConstraints();
 		expect(dbRun).not.toHaveBeenCalledWith(
-			'CREATE CONSTRAINT ON (s:Dog) ASSERT exists(s.tail)',
+			'CREATE CONSTRAINT ON (s:Dog) ASSERT exists(s.nose)',
+		);
+		expect(dbRun).toHaveBeenCalledTimes(4);
+	});
+
+	it('creates a index for canIdentify property if it doesnt exist', async () => {
+		mockConstraints({
+			stub: dbRun,
+			indexes: ['CREATE INDEX ON :Dog(nose)'],
+		});
+		schema.getTypes.mockReturnValue([
+			{ name: 'Dog', properties: { tail: { canIdentify: true } } },
+		]);
+		await initConstraints();
+		expect(dbRun).toHaveBeenCalledWith('CREATE INDEX ON :Dog(tail)');
+		expect(dbRun).toHaveBeenCalledTimes(5);
+	});
+
+	it('doesnt create an index for a canIdentify property if it does exist', async () => {
+		mockConstraints({
+			stub: dbRun,
+			indexes: ['CREATE INDEX ON :Dog(nose)'],
+		});
+		schema.getTypes.mockReturnValue([
+			{ name: 'Dog', properties: { nose: { canIdentify: true } } },
+		]);
+		await initConstraints();
+		expect(dbRun).not.toHaveBeenCalledWith('CREATE INDEX ON :Dog(nose)');
+
+		expect(dbRun).toHaveBeenCalledTimes(4);
+	});
+
+	it('doesnt create an index if a uniqueness constraint will be created', async () => {
```

Comment:
I don't really understand this description, is this interpretation correct?:

> does not create an index for fields with an existing unique constraint

Reply:
Yep - your interpretation is much easier to understand so I will update that too. Thanks